package/firmware/linux-firmware/intel.mk

```diff
@@ -216,29 +216,49 @@ define Package/e100-firmware/install
 endef
 $(eval $(call BuildPackage,e100-firmware))
 
-Package/intel-igpu-firmware-dmc = $(call Package/firmware-default,Intel iGPU DMC Display MC firmware)
-define Package/intel-igpu-firmware-dmc/install
+i915_deps:=+i915-firmware-dmc +i915-firmware-guc +i915-firmware-huc +i915-firmware-gsc
+Package/i915-firmware = $(call Package/firmware-default,Intel I915 firmware \(meta package\),$(i915_deps),LICENSE.i915)
+define Package/i915-firmware/install
+	true
+endef
+$(eval $(call BuildPackage,i915-firmware))
+
+Package/i915-firmware-dmc = $(call Package/firmware-default,Intel I915 DMC firmware,,LICENSE.i915)
+define Package/i915-firmware-dmc/install
 	$(INSTALL_DIR) $(1)/lib/firmware/i915
-	$(CP) \
-		$(PKG_BUILD_DIR)/i915/*_dmc_*.bin* \
-		$(1)/lib/firmware/i915/
+	for f in $(PKG_BUILD_DIR)/i915/*_dmc*.bin; do                        \
+	  t=`echo $$$${f##*/} | cut -d_ -f2 | cut -d. -f1`;                  \
+	  if [ "$$$$t" = dmc ]; then $(CP) $$$$f $(1)/lib/firmware/i915/; fi \
+	done
 endef
-$(eval $(call BuildPackage,intel-igpu-firmware-dmc))
+$(eval $(call BuildPackage,i915-firmware-dmc))
 
-Package/intel-igpu-firmware-guc = $(call Package/firmware-default,Intel iGPU GUC Graphics MC firmware)
-define Package/intel-igpu-firmware-guc/install
+Package/i915-firmware-guc = $(call Package/firmware-default,Intel I915 GUC firmware,,LICENSE.i915)
+define Package/i915-firmware-guc/install
 	$(INSTALL_DIR) $(1)/lib/firmware/i915
-	$(CP) \
-		$(PKG_BUILD_DIR)/i915/*_guc_*.bin* \
-		$(1)/lib/firmware/i915/
+	for f in $(PKG_BUILD_DIR)/i915/*_guc*.bin; do                        \
+	  t=`echo $$$${f##*/} | cut -d_ -f2 | cut -d. -f1`;                  \
+	  if [ "$$$$t" = guc ]; then $(CP) $$$$f $(1)/lib/firmware/i915/; fi \
+	done
 endef
-$(eval $(call BuildPackage,intel-igpu-firmware-guc))
+$(eval $(call BuildPackage,i915-firmware-guc))
 
-Package/intel-igpu-firmware-huc = $(call Package/firmware-default,Intel iGPU HUC H.265 MC firmware)
-define Package/intel-igpu-firmware-huc/install
+Package/i915-firmware-huc = $(call Package/firmware-default,Intel I915 HUC firmware,,LICENSE.i915)
+define Package/i915-firmware-huc/install
 	$(INSTALL_DIR) $(1)/lib/firmware/i915
-	$(CP) \
-		$(PKG_BUILD_DIR)/i915/*_huc_*.bin* \
-		$(1)/lib/firmware/i915/
+	for f in $(PKG_BUILD_DIR)/i915/*_huc*.bin; do                        \
+	  t=`echo $$$${f##*/} | cut -d_ -f2 | cut -d. -f1`;                  \
+	  if [ "$$$$t" = huc ]; then $(CP) $$$$f $(1)/lib/firmware/i915/; fi \
```

Comment:
why do you need this for loop here?

Reply:
The file name pattern in regex is `([[:alnum:]]+)_([[:alnum:]]+)(_\w+)?`, where `$1` is the platform name, `$2` is the firmware type (such as dmc, guc, huc, etc.), and the optional $3 which is revision or other suffix.
It is hard to handle such pattern with glob mechanism, so my solution is to glob it first to narrow down the target file set, and then split with `_` to extract the firmware type (remove the ".bin" in case there is no $3, e.g. "adlp_dmc.bin")

Reply:
Linux-firmware has these huc files:
```
i915/bxt_huc_2.0.0.bin
i915/bxt_huc_ver01_07_1398.bin
i915/bxt_huc_ver01_8_2893.bin
i915/cml_huc_4.0.0.bin
i915/dg1_huc_7.7.1.bin
i915/dg1_huc_7.9.3.bin
i915/dg1_huc.bin
i915/dg2_huc_gsc.bin
i915/ehl_huc_9.0.0.bin
i915/glk_huc_4.0.0.bin
i915/glk_huc_ver03_01_2893.bin
i915/icl_huc_9.0.0.bin
i915/icl_huc_ver8_4_3238.bin
i915/kbl_huc_4.0.0.bin
i915/kbl_huc_ver02_00_1810.bin
i915/mtl_huc_gsc.bin
i915/skl_huc_2.0.0.bin
i915/skl_huc_ver01_07_1398.bin
i915/tgl_huc_7.0.12.bin
i915/tgl_huc_7.0.3.bin
i915/tgl_huc_7.5.0.bin
i915/tgl_huc_7.9.3.bin
i915/tgl_huc.bin
```
Which of them do you want to package here?

Reply:
All of these will be in the i915-firmware-huc, especially tgl_huc.bin and dg2_huc_gsc.bin

Reply:
why do you need this change?

Reply:
> why do you need this change?

Can you be more specific?

If you are asking why need to classify the i915 firmwares into separated packages, check the [previous discussion](https://github.com/openwrt/openwrt/pull/16276#discussion_r1758520927)

If you are talking about the "for loop", here is my answer:

> The for loop is especally important to handle `mtl_huc_gsc.bin` correctly. Without it both `*_huc*.bin` and `*_gcs*.bin` will match it, which means both `i915-firmware-huc` and `i915-firmware-gsc` packages will contain `mtl_huc_gsc.bin`, it causes build error

More details in commit message of [linux-firmware: rename packages for i915 firmware](https://github.com/openwrt/openwrt/pull/16276/commits/63a93f04689934c627d9de40601bcea17911817b)